pkg/controlplane/apiserver/aggregator.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiserver
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// APIServicePriority defines group priority that is used in discovery. This controls
+// group position in the kubectl output.
+type APIServicePriority struct {
+	// Group indicates the order of the group relative to other groups.
+	Group int32
+	// Version indicates the relative order of the Version inside of its group.
+	Version int32
+}
+
+// DefaultGenericAPIServicePriorities returns the APIService priorities for generic APIs
+func DefaultGenericAPIServicePriorities() map[schema.GroupVersion]APIServicePriority {
+	// The proper way to resolve this letting the aggregator know the desired group and version-within-group order of the underlying servers
+	// is to refactor the genericapiserver.DelegationTarget to include a list of priorities based on which APIs were installed.
+	// This requires the APIGroupInfo struct to evolve and include the concept of priorities and to avoid mistakes, the core storage map there needs to be updated.
+	// That ripples out every bit as far as you'd expect, so for 1.7 we'll include the list here instead of being built up during storage.
+	return map[schema.GroupVersion]APIServicePriority{
+		{Group: "", Version: "v1"}: {Group: 18000, Version: 1},
+		// to my knowledge, nothing below here collides
+		{Group: "events.k8s.io", Version: "v1"}:                      {Group: 17750, Version: 15},
+		{Group: "events.k8s.io", Version: "v1beta1"}:                 {Group: 17750, Version: 5},
+		{Group: "authentication.k8s.io", Version: "v1"}:              {Group: 17700, Version: 15},
+		{Group: "authentication.k8s.io", Version: "v1beta1"}:         {Group: 17700, Version: 9},
+		{Group: "authentication.k8s.io", Version: "v1alpha1"}:        {Group: 17700, Version: 1},
+		{Group: "authorization.k8s.io", Version: "v1"}:               {Group: 17600, Version: 15},
+		{Group: "autoscaling", Version: "v1"}:                        {Group: 17500, Version: 15},
+		{Group: "autoscaling", Version: "v2"}:                        {Group: 17500, Version: 30},
+		{Group: "autoscaling", Version: "v2beta1"}:                   {Group: 17500, Version: 9},
+		{Group: "autoscaling", Version: "v2beta2"}:                   {Group: 17500, Version: 1},
```

Comment:
Any reason these are in the controlplane code  and not in kube-apiserver?

Reply:
aren't these for kube-apiserver related aka workloads?

Reply:
Aren't they working against any resource with scale resource? Then they are generic.

Reply:
I think I agree with you, looking into that group which resources it exposes. Updated the PR.